doc/banner/banner_hld.md

```diff
@@ -0,0 +1,233 @@
+    2. Logout message
+    3. MOTD
+
+# 2 Design
+
+## 2.1 Overview
+Here is the representation of SONiC platform using Banner feature:
+
+![Banner system chart diagram](images/banner_system_chart_diagram.png "Figure 1: Banner system chart diagram")
+<!-- omit in toc -->
+###### Figure 1: Banner system chart diagram
+
+This feature require access to SONiC DB. All messages (MOTD, login and logout) saved into the SONiC config database. Hostcfgd will listen for the configuration changes in corresponding tables and restart banner-config service. Banner config service - it is simple SystemD service which runs before we get SSH connection. It reads configured messages from database and apply it to Linux.
+
+## 2.2 Flows
+<!-- omit in toc -->
+### 2.2.1 Banner init flow
+![Banner init flow](images/banner_init_diagram.png "Figure 2: Banner init flow")
+<!-- omit in toc -->
+###### Figure 2: Banner init flow
+
+<!-- omit in toc -->
+### 2.5.2 Banner config flow
+![Banner config flow](images/banner_config_diagram.png "Figure 3: Banner config flow")
+<!-- omit in toc -->
+###### Figure 3: Banner config flow
+
+<!-- omit in toc -->
+### 2.5.2 Banner show configuration
+![Banner show configuration](images/banner_show_diagram.png "Figure 4: Banner show configuration")
+<!-- omit in toc -->
+###### Figure 4: Banner show configuration
+
+
+## 2.3 CLI
+
+<!-- omit in toc -->
+### 2.3.1 Command structure
+
+**User interface**:
+```
+config
+\-- banner
+    |-- login <message>
+    |-- logout <message>
+    |-- motd <message>
+
+show
+\-- banner
+```
+
+**Options:**
+
+General:
+1. `<message>` - Message to be configured: `string`
+
+#### 2.3.1 Config command group
+**The following command set loin banner message:**
+```bash
+config banner login <message>
+```
+
+**The following command set logout banner message:**
+```bash
+config banner logout <message>
+```
+
+**The following command set mesage of the day (MOTD):**
+```bash
+config banner motd <message>
+```
+
+#### 2.3.2 Show command
+**The following command display banner configuration:**
+```bash
+root@sonic:/home/admin# show banner
+
+Banners:
+  Message of the Day (MOTD):
+    SONiC Switch
+
+  Login:
+    SONiC Switch Management
+
+  Logout:
+    Goodbye
+```
+
+## 2.4 YANG model
+New YANG model `sonic-banner.yang` will be added to provide support for configuring Banner messages.
+
+**Skeleton code:**
+```
+ module sonic-banner {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/sonic-banner";
+    prefix banner_message;
+
+    description "BANNER_MESSAGE YANG Module for SONiC-based OS";
+
+    revision 2023-05-18 {
+        description "First Revision";
+    }
+
+    container sonic-banner {
+
+        container BANNER_MESSAGE {
+
+            description "BANNER_MESSAGE part of config_db.json";
+
+            container pre_login {
+
+                leaf message {
+                    type string;
+                }
+
+            }
+
+            container post_login {
```

Comment:
Instead of adding 3 containers i.e pre_login, post_login & post_logout, can we have the BANNER table name and key as "global" and three 3 attributes.

Reply:
Done